docs/usage-fastapi-server.md

```diff
@@ -0,0 +1,78 @@
+# Pyserini: REST API Server with FastAPI
```

Comment:
Renamed to `usage-rest.md`? Parallels `usage-mcp.md`? I.e., focus on what it is, not the impl.